### PR TITLE
Add OffsetDateTime Java Time Generators

### PR DESCRIPTION
**Pull request by mdedetrich**
Adds OffsetDateTime instances, its basically the exact same as the `Zoned` versions but uses `OffsetDateTime.ofInstant` instead

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -38,6 +39,12 @@ def commonSettings(subProject: Option[String]): Seq[Setting[_]] = {
         Set(organization.value %% artifact.value % mimaPreviousVersion.value)
     },
 
+    mimaBinaryIssueFilters ++= Seq(
+      ProblemFilters.exclude[ReversedMissingMethodProblem](
+        "org.scalacheck.ops.time.ImplicitJavaTimeGenerators.arbOffsetDateTime"
+      )
```
Comment (jeffmay) [inline build.sbt:59]: Why is this needed? Shouldn't we just release a minor version?

Comment (mdedetrich) [reply build.sbt:59]: I did this some time ago, but iirc MIMA was complaining (`ReversedMissingMethodProblem` is meant to indicate forwards compatibility issues, which in practice is a non issue for a testing library like scalacheck-ops since you can always force the latest version).

This is just standard way that MIMA works, it warns about both backwards and forwards compatibility (i.e. if you add a public method to a public interface it will also complain).

**Review by jeffmay — APPROVED** [review]
I'll take  a look at how we can make minor changes easier with sbt-git-versioning plugin in the future. For now, this lgtm.

File: build.sbt (hunk 1 above)
Comment (jeffmay) [reply build.sbt:59]: We have an internal sbt plugin at Optum Digital (formerly Rally Health) that handles this automatically and allows you to force the snapshot versions to match the expected release version (so if you include a forwards incompatible change like this, then you have to set the `rallyVersioningSnapshotLowerBound` to the next minor version). It would then automatically ignore all forwards incompatible warnings.

The plugin was converted to an open-source project here https://github.com/rallyhealth/sbt-git-versioning, but I nobody has been maintaining that properly 😞 

Anyway, there is a significant gap in functionality between the open-source version and the internal one and I have tried a couple times to upgrade to the latest version and failed, so I kind of gave up on it. I'm potentially going to try to give it a new shot in the arm and add a better review and CI process around it.

Comment (mdedetrich) [reply build.sbt:59]: Not sure if this is relevant but the only reason I added this filter is because `mimaReportBinaryIssues` was being executed in the old travis Ci which caused the CI to fail. The newly generated github actions doesn't use `mimaReportBinaryIssues` so I can also remove this filter if you desire.

**Review by russellremple — APPROVED** [review]
